specifyweb/frontend/js_src/lib/components/AppResources/tree.ts

```diff
@@ -67,13 +67,25 @@ function getGlobalAllResources(resources: AppResources): {
   const globalDirectories = resources.directories.filter(
     (directory) => directory.scope === 'global'
   );
-  if (globalDirectories.length === 0)
-    globalDirectories.push({
-      ...addMissingFields('SpAppResourceDir', {
-        userType: 'Common',
-      }),
-      scope: 'global',
-    });
+
+  const directoriesNeeded = ['Global Prefs', 'Prefs', 'Common'];
+  if (globalDirectories.length < directoriesNeeded.length) {
+    const directoriesPresent = new Set(
+      globalDirectories.map((directory) => directory.userType)
+    );
+    const missingUserType = directoriesNeeded.find(
+      (userType) => !directoriesPresent.has(userType)
+    );
```

Comment:
this code assumes that at most 1 user type is missing - can you rewrite to add any user type that is missing, not just one?